examples/src/main/java/com/opengamma/strata/examples/finance/SwapPricingExample.java

```diff
@@ -143,49 +146,18 @@ private static void calculate(CalculationRunner runner) {
   //-----------------------------------------------------------------------  
   // create a vanilla fixed vs libor 3m swap
   private static Trade createVanillaFixedVsLibor3mSwap() {
-    NotionalSchedule notional = NotionalSchedule.of(Currency.USD, 100_000_000);
-
-    SwapLeg payLeg = RateCalculationSwapLeg.builder()
-        .payReceive(PayReceive.PAY)
-        .accrualSchedule(PeriodicSchedule.builder()
-            .startDate(LocalDate.of(2014, 9, 12))
-            .endDate(LocalDate.of(2021, 9, 12))
-            .frequency(Frequency.P6M)
-            .businessDayAdjustment(BusinessDayAdjustment.of(MODIFIED_FOLLOWING, HolidayCalendarIds.USNY))
-            .build())
-        .paymentSchedule(PaymentSchedule.builder()
-            .paymentFrequency(Frequency.P6M)
-            .paymentDateOffset(DaysAdjustment.NONE)
-            .build())
-        .notionalSchedule(notional)
-        .calculation(FixedRateCalculation.of(0.015, DayCounts.THIRTY_U_360))
-        .build();
-
-    SwapLeg receiveLeg = RateCalculationSwapLeg.builder()
-        .payReceive(PayReceive.RECEIVE)
-        .accrualSchedule(PeriodicSchedule.builder()
-            .startDate(LocalDate.of(2014, 9, 12))
-            .endDate(LocalDate.of(2021, 9, 12))
-            .frequency(Frequency.P3M)
-            .businessDayAdjustment(BusinessDayAdjustment.of(MODIFIED_FOLLOWING, HolidayCalendarIds.USNY))
-            .build())
-        .paymentSchedule(PaymentSchedule.builder()
-            .paymentFrequency(Frequency.P3M)
-            .paymentDateOffset(DaysAdjustment.NONE)
-            .build())
-        .notionalSchedule(notional)
-        .calculation(IborRateCalculation.of(IborIndices.USD_LIBOR_3M))
-        .build();
-
-    return SwapTrade.builder()
-        .product(Swap.of(payLeg, receiveLeg))
-        .info(TradeInfo.builder()
+    return FixedIborSwapConventions.USD_FIXED_6M_LIBOR_3M.toTrade(
```

Comment:
done.

On Wed, Mar 23, 2016 at 11:56 AM, Stephen Colebourne <
notifications@github.com> wrote:

> In
> examples/src/main/java/com/opengamma/strata/examples/finance/SwapPricingExample.java
> <https://github.com/OpenGamma/Strata/pull/867#discussion_r57147018>:
>
> > -            .endDate(LocalDate.of(2021, 9, 12))
> > -            .frequency(Frequency.P3M)
> > -            .businessDayAdjustment(BusinessDayAdjustment.of(MODIFIED_FOLLOWING, HolidayCalendarIds.USNY))
> > -            .build())
> > -        .paymentSchedule(PaymentSchedule.builder()
> > -            .paymentFrequency(Frequency.P3M)
> > -            .paymentDateOffset(DaysAdjustment.NONE)
> > -            .build())
> > -        .notionalSchedule(notional)
> > -        .calculation(IborRateCalculation.of(IborIndices.USD_LIBOR_3M))
> > -        .build();
> > -
> > -    return SwapTrade.builder()
> > -        .product(Swap.of(payLeg, receiveLeg))
> > -        .info(TradeInfo.builder()
> > +    return FixedIborSwapConventions.USD_FIXED_6M_LIBOR_3M.toTrade(
>
> Suggest using a local variable for tradeInfo. The other parameters should
> have an end of line comment to explain what they are.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenGamma/Strata/pull/867/files/ea76d3320d03f0eee9321d655d65060d0c2dbab5#r57147018>
>